src/Task/Paratest.php

```diff
@@ -0,0 +1,104 @@
+                'debugger'       => null,
+                'coverage-xml'   => null,
+                'coverage-html'  => null,
+                'log-junit'      => null,
+                'testsuite'      => null,
+                'config'         => null,
+            ]
+        );
+
+        $resolver->addAllowedTypes('processes', ['null', 'integer', 'string']);
+        $resolver->addAllowedTypes('functional', ['bool']);
+        $resolver->addAllowedTypes('phpunit', ['null', 'string']);
+        $resolver->addAllowedTypes('configuration', ['null', 'string']);
+        $resolver->addAllowedTypes('always_execute', ['bool']);
+        $resolver->addAllowedTypes('runner', ['null', 'string']);
+        $resolver->addAllowedTypes('debugger', ['null', 'array']);
+        $resolver->addAllowedTypes('coverage-xml', ['null', 'string']);
+        $resolver->addAllowedTypes('coverage-html', ['null', 'string']);
+        $resolver->addAllowedTypes('log-junit', ['null', 'string']);
+        $resolver->addAllowedTypes('testsuite', ['null', 'string']);
+
+        return $resolver;
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return $context instanceof GitPreCommitContext || $context instanceof RunContext;
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfiguration();
+
+        $files = $context->getFiles()->name('*.php');
+        if (0 === \count($files) && !$config['always_execute']) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('paratest');
+        $arguments->addOptionalArgument('-p %s', $config['processes']);
+        $arguments->addOptionalArgument('-f', $config['functional']);
+        $arguments->addOptionalArgument('-c %s', $config['configuration']);
+        $arguments->addOptionalArgument('--phpunit %s', $config['phpunit']);
+        $arguments->addOptionalArgument('--runner %s', $config['runner']);
+        $arguments->addOptionalArgument('--coverage-xml %s', $config['coverage-xml']);
+        $arguments->addOptionalArgument('--coverage-html %s', $config['coverage-html']);
+        $arguments->addOptionalArgument('--log-junit %s', $config['log-junit']);
+        $arguments->addOptionalArgument('--testsuite %s', $config['testsuite']);
+        $arguments->addOptionalCommaSeparatedArgument('--group=%s', $config['group']);
+
+        $coverageEnabled = function ($config) {
+            return !empty($config['coverage-xml'])
+                || !empty($config['coverage-html'])
+                || !empty($config['log-junit']);
+        };
+        if ($coverageEnabled($config) && !empty($config['debugger'])) {
+            $bin = $config['debugger']['bin'];
+            $args = $config['debugger']['args'];
+            $arguments = new ProcessArgumentsCollection(array_merge([$bin], $args, $arguments->toArray()));
```

Comment:
don't you need to `...$args, ...$arguments->toArray()` ?

Reply:
I don't think so? The array merge should handle it to become a single array, and that is what the constructor expects.